homeassistant/auth_providers/modules/insecure_example.py

```diff
@@ -0,0 +1,66 @@
+"""Example auth module."""
```

Comment:
This should not live under auth provider as it has nothing to do with it? 

I suggest we restructure our auth into

- `homeassistant/auth/__init__.py`
- `homeassistant/auth/providers/homeassistant.py`
- `homeassistant/auth/modules/totp.py`

I also not sure if I like `modules`, it's too generic. What about `two_factor` ?